controllers/infinispan_controller.go

```diff
@@ -1760,18 +1763,32 @@ func GossipRouterPodList(infinispan *infinispanv1.Infinispan, kube *kube.Kuberne
 	return podList, kube.ResourcesList(infinispan.Namespace, GossipRouterPodLabels(infinispan.Name), podList, ctx)
 }
 
-func buildStartupArgs(overlayConfigMapKey string, zeroCapacity string) []string {
-	var args []string
+func buildStartupArgs(overlayConfigMapKey, overlayLog4jConfigMapKey, zeroCapacity string) []string {
+	var args strings.Builder
+
+	// Preallocate a buffer of 10 bytes to speed up string building (saves code from growing the memory dynamically)
+	args.Grow(110)
+	args.WriteString("-Dinfinispan.zero-capacity-node=" + zeroCapacity)
```

Comment:
This should be two different calls to `WriteString`. When using the builder you shouldn't use `+` at all.

e.g.

```go
args.WriteStritng("-Dinfinispan.zero-capacity-node=")
args.WriteString(zeroCapacity)
```